test/suites/dev/test-xcm-v2/test-mock-hrmp-asset-transfer-5.ts

```diff
@@ -0,0 +1,150 @@
+import "@moonbeam-network/api-augment";
+import { beforeAll, describeSuite, expect } from "@moonwall/cli";
+
+import { BN, u8aToHex } from "@polkadot/util";
+import { ParaId } from "@polkadot/types/interfaces";
+import { alith, baltathar } from "@moonwall/util";
+import { XcmFragment, RawXcmMessage, injectHrmpMessageAndSeal } from "../../../helpers/xcm.js";
+
+import { expectOk } from "../../../helpers/expect.js";
+
+const foreign_para_id = 2000;
+
+describeSuite({
+  id: "D3412",
+  title: "Mock XCM - receive horizontal transfer",
+  foundationMethods: "dev",
+  testCases: ({ context, it, log }) => {
+    let assetId: string;
+    let paraId: ParaId;
+    let transferredBalance: bigint;
+    let sovereignAddress: string;
+
+    beforeAll(async () => {
+      // registerAsset
+      const { result } = await context.createBlock(
+        context
+          .polkadotJs()
+          .tx.sudo.sudo(
+            context
+              .polkadotJs()
+              .tx.assetManager.registerLocalAsset(
+                baltathar.address,
+                baltathar.address,
+                true,
+                new BN(1)
+              )
+          )
+      );
+
+      const eventsRegister = result?.events;
+
+      // Look for assetId in events
+      assetId = eventsRegister!
+        .find(({ event: { section } }) => section.toString() === "assetManager")
+        .event.data[0].toHex()
+        .replace(/,/g, "");
```

Comment:
Converting events to strings and doing find/replace makes me feel uneasy. Consider instead using the previous methods:
```
      const event = eventsRegister!.find(({ event }) => context.polkadotJs().events.assetManager.LocalAssetRegistered.is(event))!
       assetId = event.event.data.assetId.toHex()
```